File: DemoApp Objective-C/CASSample/AppOpenAd/AppOpenVC.m
//
//  AppOpenVC.m
//  CASSample
//

#import "AppOpenVC.h"
#import "AppDelegate.h"

@interface AppOpenVC ()

@property (strong, nonatomic) CASAppOpen *appOpen;

@end

@implementation AppOpenVC

- (void)viewDidLoad {
    [super viewDidLoad];

    self.appOpen = [[CASAppOpen alloc] initWithCasID: [AppDelegate casId]];
    self.appOpen.delegate = self;
    self.appOpen.impressionDelegate = self;
    
    self.navigationItem.largeTitleDisplayMode = UINavigationItemLargeTitleDisplayModeNever;
    self.navigationItem.title = @"App Open Ad";
}

- (IBAction)loadAction:(UIButton *)sender {
    [self.appOpen loadAd];
}

- (IBAction)showAction:(UIButton *)sender {
    [self.appOpen presentFromViewController:self];
}

#pragma mark - CASImpressionDelegate

- (void)adDidRecordImpressionWithInfo:(CASContentInfo *)info {
    NSLog(@"adDidRecordImpressionWithInfo");
}

#pragma mark - CASScreenContentDelegate

- (void)screenAdDidLoadContent:(id<CASScreenContent>)ad {
    NSLog(@"screenAdDidLoadContent");
}

- (void)screenAd:(id<CASScreenContent>)ad didFailToLoadWithError:(CASError *)error {
    NSLog(@"screenAd:didFailToLoadWithError %@", error.description);
}

- (void)screenAdWillPresentContent:(id<CASScreenContent>)ad {
    NSLog(@"screenAdWillPresentContent");
}

- (void)screenAd:(id<CASScreenContent>)ad didFailToPresentWithError:(CASError *)error {
    NSLog(@"screenAd:didFailToPresentWithError %@", error.description);
}

- (void)screenAdDidClickContent:(id<CASScreenContent>)ad {
    NSLog(@"screenAdDidClickContent");
}

- (void)screenAdDidDismissContent:(id<CASScreenContent>)ad {
    NSLog(@"screenAdDidDismissContent");
}

@end

File: DemoApp Swift/CASSample/AppOpenAd/AppOpenVC.swift
//
//  AppOpenVC.swift
//  CASSample
//
//  Copyright © 2025 Clever Ads Solutions. All rights reserved.
//

import UIKit
import CleverAdsSolutions

class AppOpenVC: UIViewController {
                    
    @IBOutlet var loadButton: UIButton!
    @IBOutlet var showButton: UIButton!
            
    private var appOpen = CASAppOpen(casID: AppDelegate.casId)

    override func viewDidLoad() {
        super.viewDidLoad()
        appOpen.delegate = self
        appOpen.impressionDelegate = self
        //
        navigationItem.largeTitleDisplayMode = .never
        navigationItem.title = "App Open Ad"
    }
    
    @IBAction private func loadAction(_ sender: UIButton) {
        appOpen.loadAd()
    }
    
    @IBAction private func showAction(_ sender: UIButton) {
        appOpen.present(from: self)
    }
}

extension AppOpenVC: CASImpressionDelegate {
    func adDidRecordImpression(info: AdContentInfo) {
        print(#function, "Ad Source: \(info.sourceName)")
    }
}

extension AppOpenVC: CASScreenContentDelegate {
    func screenAdDidLoadContent(_ ad: any CASScreenContent) {
        print(#function)
    }
    
    func screenAd(_ ad: any CASScreenContent, didFailToLoadWithError error: AdError) {
        print(#function, "Error: \(error.description)")
    }
    
    func screenAdWillPresentContent(_ ad: any CASScreenContent) {
        print(#function)
    }
    
    func screenAd(_ ad: any CASScreenContent, didFailToPresentWithError error: AdError) {
        print(#function, "Error: \(error.description)")
    }
    
    func screenAdDidClickContent(_ ad: any CASScreenContent) {
        print(#function)
    }
    
    func screenAdDidDismissContent(_ ad: any CASScreenContent) {
        print(#function)
    }
}
